### ISSUE TITLE
exec: "/bin/sh": stat /bin/sh: no such file or directory

### ISSUE THREAD

**Issue by jmreicha** [YungSang/boot2docker-vagrant-box#11]
I am receiving the following error if I attempt to pull an image with many layers from the dockerhub.

```
exec: "/bin/sh": stat /bin/sh: no such file or directory
```

Less than that, no problem.

From my research so far this seems to be an issue with custom compiled kernels and some sort of limitations set on the number of layers allowed by AUFS, https://github.com/docker/docker/issues/5135.

Do you know of any workaround for this?

**Comment by YungSang** [comment]
Unfortunately AUFS has the limitations.
In this case I use CoreOS instead of boot2docker, though CoreOS doesn't have VirtualBox Guest Additions.
Please try `yungsang/coreos`, or `yungsang/coreos-alpha` for docker v1.3.0.
https://github.com/YungSang/coreos-packer
https://vagrantcloud.com/yungsang

**Comment by YungSang** [comment]
I got an idea.
https://github.com/boot2docker/boot2docker/blob/master/doc/WORKAROUNDS.md#btrfs-ie-mkfs-inside-a-privileged-container

I am trying to format boot2docker-data with btrfs.

**Comment by YungSang** [comment]
https://github.com/YungSang/boot2docker-vagrant-box/releases/tag/btrfs-v1.3.1

Please try it.

```
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.box = "boot2docker-btrfs"

  config.vm.box_url = "https://github.com/YungSang/boot2docker-vagrant-box/releases/download/btrfs-v1.3.1/boot2docker-virtualbox.box"
end
```

**Comment by YungSang** [comment]
42 layer image test
https://gist.github.com/harto/10610004
Passed.

**Comment by YungSang** [comment]
I created devicemapper version as well.
https://github.com/YungSang/boot2docker-vagrant-box/releases/tag/devicemapper-v1.3.1

```
VAGRANTFILE_API_VERSION = "2"

Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
  config.vm.box = "boot2docker-devicemapper"

  config.vm.box_url = "https://github.com/YungSang/boot2docker-vagrant-box/releases/download/devicemapper-v1.3.1/boot2docker-virtualbox.box"
end
```

**Comment by jmreicha** [comment]
Awesome, I will test this out today.

**Comment by YungSang** [comment]
@jmreicha Thank you so much for giving me opportunity to do this.

**Comment by jmreicha** [comment]
@YungSang Thank you!

I am testing now,  everything looks good so far inside the vagrant box, no 42 layer issues.  Yay.

I did notice a tls error outside of the VM after bringing up the btrfs box though.  I haven't started troubleshooting it yet but thought I should at least mention it.

```
docker ps                                                                                                                                                 
2014/10/31 11:07:26 Get https://localhost:2375/v1.15/containers/json: tls: oversized record received with length 20527
```

Any thoughts there?

**Comment by jmreicha** [comment]
Okay I don't know what happened or changed but I am not experiencing the issue posted above now, I wish I knew what happened there.  Thanks again for the help!

**Comment by YungSang** [comment]
@jmreicha Thank you for testing.

I just changed a storage for docker daemon, so it shouldn't happen.
No issue at my side.

Could you tell me which version of docker in your local machine and Vagrantfile you use?
And I would like to know your procedure exactly to reproduce it.
How about devicemapper version or normal version in the same procedure?

**Comment by YungSang** [comment]
It looks like when `docker --tls ps` is executed.
I guess that you have sort of TLS settings for the docker client.
But my boot2docker boxes don't have TLS capability.

**Comment by jmreicha** [comment]
Yeah I can't reproduce so I am not going to worry about it.  Thanks!

**Comment by YungSang** [comment]
@jmreicha That's good to know. :)
You're welcome.

**Comment by YungSang** [comment]
@jmreicha I found another workaround there.
https://github.com/boot2docker/boot2docker/issues/531
https://github.com/mitchellh/boot2docker-vagrant-box/pull/72

I've just released v1.3.2. :)
